apps/os/backend/agent/mcp/mcp-event-hooks.ts

```diff
@@ -260,10 +271,46 @@ export async function handleMCPConnectRequest(
     return events;
   }
 
+  const db = getDb();
+  const auth = getAuth(db);
+
+  const agentClassName = agentDurableObjectName.startsWith("SlackAgent-")
+    ? "SlackAgent"
+    : "IterateAgent";
+
+  // Get the organization ID for the estate to construct proper callback URL
+  let organizationId: string | undefined;
+  if (!finalRedirectUrl) {
+    try {
+      const estateWithOrg = await db.query.estate.findFirst({
```

Comment:
FYI according to cloudflare it takes up to 30ms to create the DB connection before the first query - I don't have enough context to know whether this matters here